### PR TITLE
Update value-types-and-reference-types.md

### LINKED ISSUE #13180: Explaining reference types with pointers
The page explains with regard to reference types:

> A reference type contains a pointer to another memory location that holds the data.

Some years back, Eric Lippert made a strong argument why references should not be explained as 
(memory) addresses. Link to blog post: https://blogs.msdn.microsoft.com/ericlippert/2009/02/17/references-are-not-addresses/

Sure the documentation page says a reference type _contains_ a pointer, not that a reference type _is_ a pointer. But often, readers will simply interpret this explanation to mean a reference type _is_ a pointer.  

In the end, how a reference is implemented close to the metal is an implementation detail of the runtime/JITter and shouldn't really be addressed (pun intended) on this documentation page, as trying to do so in an overly simplified manner would just invite misunderstandings and misconceptions about references, imo.

Also noteworthy is the C# counterpart of this page (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types). It avoids talking about references having/containing pointers or addresses entirely. This VB.NET page here should perhaps be organized in a similar manner as its C# counterpart.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9f0e0036-37f7-9683-1a90-8700577b28dc
* Version Independent ID: 4ea79c95-78f2-4c2a-3070-afb6de4b7dbb
* Content: [Value Types and Reference Types](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types)
* Content Source: [docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md](https://github.com/dotnet/docs/blob/master/docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md)
* Product: **dotnet-visualbasic**
* GitHub Login: @KathleenDollard
* Microsoft Alias: **kdollard**

### PR DESCRIPTION
**Pull request by KathleenDollard**
## Summary

Describe your changes here.

Fixes #13180

### CODE REVIEW

File: docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ helpviewer_keywords:
 ms.assetid: fc82ce15-5a40-4c5c-a1e1-a556830e7391
 ---
 # Value Types and Reference Types
-In Visual Basic, data types are implemented based on their classification. The Visual Basic data types can be classified according to whether a variable of a particular type stores its own data or a pointer to the data. If it stores its own data it is a *value type*; if it holds a pointer to data elsewhere in memory it is a *reference type*.  
+There are two kinds of types in Visual Basic: reference types and value types. Variables of reference types store references to their data (objects), while variables of value types directly contain their data. With reference types, two variables can reference the same object; therefore, operations on one variable can affect the object referenced by the other variable. With value types, each variable has its own copy of the data, and it is not possible for operations on one variable to affect the other (except in the case of [ByRef modifier on parameters](https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/modifiers/byref)).
```
Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md]:
```suggestion
There are two kinds of types in Visual Basic: reference types and value types. Variables of reference types store references to their data (objects), while variables of value types directly contain their data. With reference types, two variables can reference the same object; therefore, operations on one variable can affect the object referenced by the other variable. With value types, each variable has its own copy of the data, and it is not possible for operations on one variable to affect the other (except in the case of [ByRef modifier on parameters](../../../language-reference/modifiers/byref)).
```

Two suggestions here: One to remove the `en-us` locale provider. The second to change to a relative link, which provides build validation for the link.

**Review by BillWagner — APPROVED** [review]
The content looks great @KathleenDollard 

After you update for the one suggestion I made, you can :shipit:

File: docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md (hunk 1 above)
Comment (rpetrusha) [reply docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md]: One additional suggestion: precede ByRef with "the".

File: docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ helpviewer_keywords:
 ms.assetid: fc82ce15-5a40-4c5c-a1e1-a556830e7391
 ---
 # Value Types and Reference Types
-In Visual Basic, data types are implemented based on their classification. The Visual Basic data types can be classified according to whether a variable of a particular type stores its own data or a pointer to the data. If it stores its own data it is a *value type*; if it holds a pointer to data elsewhere in memory it is a *reference type*.  
+There are two kinds of types in Visual Basic: reference types and value types. Variables of reference types store references to their data (objects), while variables of value types directly contain their data. With reference types, two variables can reference the same object; therefore, operations on one variable can affect the object referenced by the other variable. With value types, each variable has its own copy of the data, and it is not possible for operations on one variable to affect the other (except in the case of [ByRef modifier on parameters](../../../language-reference/modifiers/byref)).
```
Comment (KathleenDollard) [inline docs/visual-basic/programming-guide/language-features/data-types/value-types-and-reference-types.md]:
```suggestion
There are two kinds of types in Visual Basic: reference types and value types. Variables of reference types store references to their data (objects), while variables of value types directly contain their data. With reference types, two variables can reference the same object; therefore, operations on one variable can affect the object referenced by the other variable. With value types, each variable has its own copy of the data, and it is not possible for operations on one variable to affect the other (except in the case of the [ByRef modifier on parameters](../../../language-reference/modifiers/byref)).
```

**Comment by KathleenDollard** [comment]
Both comments fixed. Let me know if I need to o anything else.

**Pull request merged**